concordium-consensus/src/Concordium/External.hs

```diff
@@ -1000,381 +995,16 @@ toCString lbs = do
-    res <- runMVR a mvr
-    jsonCString res
-
--- | Decode a block hash from a null-terminated base-16 string.
-decodeBlockHash :: CString -> IO (Maybe BlockHash)
-decodeBlockHash blockcstr = readMaybe <$> peekCString blockcstr
-
--- | Decode an account address from a null-terminated base-58 string.
-decodeAccountAddress :: CString -> IO (Either String AccountAddress)
-decodeAccountAddress acctstr = addressFromBytes <$> BS.packCString acctstr
-
--- | Decode an instance address from a null-terminated JSON-encoded string.
-decodeInstanceAddress :: CString -> IO (Maybe ContractAddress)
-decodeInstanceAddress inststr = AE.decodeStrict <$> BS.packCString inststr
-
--- | Decode a module reference from a null-terminated base-16 string.
-decodeModuleRef :: CString -> IO (Maybe ModuleRef)
-decodeModuleRef modstr = readMaybe <$> peekCString modstr
-
--- | Decode the context passed to the @invokeContract@ method.
-decodeContractContext :: CString -> IO (Maybe InvokeContract.ContractContext)
-decodeContractContext ctxStr = AE.decodeStrict <$> BS.packCString ctxStr
-
--- | Decode a transaction hash from a null-terminated base-16 string.
-decodeTransactionHash :: CString -> IO (Maybe TransactionHash)
-decodeTransactionHash trHashStr = readMaybe <$> peekCString trHashStr
-
--- ** General queries
-
--- | Returns a null-terminated string with a JSON representation of the current status of Consensus.
-getConsensusStatus :: StablePtr ConsensusRunner -> IO CString
-getConsensusStatus cptr = jsonQuery cptr Q.getConsensusStatus
-
--- ** Queries against latest tree
-
--- | Returns a null-terminated string with a JSON representation of the current branches from the
---  last finalized block (inclusive).
-getBranches :: StablePtr ConsensusRunner -> IO CString
-getBranches cptr = jsonQuery cptr Q.getBranches
-
--- | Get the list of live blocks at a given height.
---  The height is interpreted relative to the genesis block at the specified index.
---  The last parameter indicates whether to restrict to a single genesis (if it is a non-zero value).
---  Returns a null-terminated string encoding a JSON list.
-getBlocksAtHeight ::
-    StablePtr ConsensusRunner ->
-    -- | Block height to query
-    Word64 ->
-    -- | Genesis index that block height is based on
-    Word32 ->
-    -- | Non-zero to restrict to blocks at specified genesis index
-    Word8 ->
-    IO CString
-getBlocksAtHeight cptr height genIndex restrict =
-    jsonQuery cptr $
-        fmap fst
-            <$> Q.getBlocksAtHeight
-                (BlockHeight height)
-                (GenesisIndex genIndex)
-                (restrict /= 0)
-
--- | Retrieve the last finalized block height relative to the most recent genesis index. Used for
--- resuming out-of-band catchup.
-getLastFinalizedBlockHeight ::
-    StablePtr ConsensusRunner ->
-    IO Word64
-getLastFinalizedBlockHeight cptr = do
-    (ConsensusRunner mvr) <- deRefStablePtr cptr
-    theBlockHeight <$> runMVR Q.getLastFinalizedBlockHeight mvr
-
```

Comment:
It seems this function is still used in `ffi.rs`.

Reply:
Yes, indeed.